datahub/oauth/admin/middleware.py

```
@@ -0,0 +1,31 @@
+from time import time
+
+from django.contrib.auth import logout as django_logout
+from django.shortcuts import redirect
+from rest_framework.reverse import reverse
+
+
+class OAuthSessionMiddleware:
+    """
+    OAuthSessionMiddleware checks if user has been logged in via OAuth.
+    """
+
+    def __init__(self, get_response):
+        """Get response."""
+        self.get_response = get_response
+
+    def __call__(self, request):
+        """
+        Check if OAuth2 expiration time is present in the session.
+
+        If OAuth2 expires on has passed it means we have to logout the user.
+        """
+        if hasattr(request, 'user') and request.user.is_authenticated:
```

It looks like `AuthenticationMiddleware` sets `request.user` unconditionally to me – I think we should avoid `hasattr(request, 'user')` since that would silently skip the check if the middleware are reordered.

---

That's a very good point. I have removed this bit now. Thank you!